src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -135,14 +171,20 @@ object GenomeReference {
   def gen: Gen[GenomeReference] = for {
     name <- Gen.identifier
     nContigs <- Gen.choose(3, 50)
-    contigs <- Gen.distinctBuildableOfN[Array, Contig](nContigs, Contig.gen)
-    contigNames = contigs.map(_.name).toSet
-    xContig <- Gen.oneOfSeq(contigNames.toSeq)
-    yContig <- Gen.oneOfSeq((contigNames - xContig).toSeq)
-    mtContig <- Gen.oneOfSeq((contigNames - xContig - yContig).toSeq)
+    contigs <- Gen.distinctBuildableOfN[Array, String](nContigs, Gen.identifier)
+    lengths <- Gen.distinctBuildableOfN[Array, Int](nContigs, Gen.choose(1000000, 500000000))
+    xContig <- Gen.oneOfSeq(contigs)
+    yContig <- Gen.oneOfSeq((contigs.toSet - xContig).toSeq)
+    mtContig <- Gen.oneOfSeq((contigs.toSet - xContig - yContig).toSeq)
```

Comment:
Should we generally check/enforce that the X, Y, and MT contigs are distinct? You're doing that in the test. It seems reasonable to disallow unless someone comes along with a compelling reason to allow them to coincide.